### PR TITLE
Fixed Reports page navigation and re-login

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#9199](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9199)

* [ ] Enhancement
* [x] Bug
* [ ] Documentation

## Description

Service request to inversify for components happen inside the mount or create steps.
Reordered variables and fixed initialization of objects.
The actual issue was happening on the Reports page, all other changes fixed to avoid other issues.

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [x] Not Required

### Steps to Reproduce

On the timeline, go to reports, navigate to another page (profile or timeline). Check that the cookie storage is not blank

### UI Changes

NO

## Notes/Additional Comments

The root cause of this issue is bad initialization and a possible issue on the cookie-storage library we are using. Logged an issue on their page as well.

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/views/reports.vue [hunk 1]
```diff
@@ -129,20 +129,10 @@ import { IAuthenticationService } from "@/services/interfaces";
     },
 })
 export default class ReportsView extends Vue {
-    private authenticationService: IAuthenticationService = container.get(
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/views/reports.vue:132]: This line was causing the loggin issue

**Review by sslaws — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/components/navmenu/navHeader.vue [hunk 2]
```diff
@@ -134,12 +133,17 @@ export default class HeaderComponent extends Vue {
...
 
     private sidebarExpanded: boolean = false;
+    private logger!: ILogger;
 
     @Watch("$route")
     private closeSidebar() {
         this.sidebarExpanded = false;
     }
 
+    private mouned() {
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/navmenu/navHeader.vue:143]: mouned --> mounted

File: Apps/WebClient/src/ClientApp/src/views/reports.vue (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/views/reports.vue:132]: wow nice found, tricky one. Mbe our best practice is to initialize these in the mounted method.

**Review by leobcgov — CHANGES_REQUESTED** [review]
typo issue

File: Apps/WebClient/src/ClientApp/src/components/navmenu/navHeader.vue (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/navmenu/navHeader.vue:143]: good catch!

**Review by leobcgov — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**